pubtools/_quay/push_docker.py

```diff
@@ -538,49 +539,53 @@ def run(self):
-            sig_remover.set_quay_client(self.dest_quay_client)
-
-            container_signature_handler.sign_container_images(docker_push_items)
-            # Push container images
-            container_pusher = ContainerImagePusher(docker_push_items, self.target_settings)
-            container_pusher.push_container_images()
-
-            if operator_push_items:
-                # Build index images
-                operator_pusher = OperatorPusher(operator_push_items, self.target_settings)
-                existing_index_images = operator_pusher.get_existing_index_images(
-                    self.dest_quay_client
-                )
-                iib_results = operator_pusher.build_index_images()
-                # Sign operator images
-                operator_signature_handler.sign_operator_images(iib_results, index_stamp)
-                # Push index images to Quay
-                operator_pusher.push_index_images(iib_results, index_stamp)
-        except (Exception, SystemExit):
-            LOG.error("An exception has occurred during the push, starting rollback")
-            self.rollback(backup_tags, rollback_tags)
-            raise
-        else:
-            # Remove old signatures
-            self.remove_old_signatures(
-                docker_push_items,
-                operator_push_items,
-                existing_index_images,
-                iib_results,
-                backup_tags,
-                container_signature_handler,
-                operator_signature_handler,
-                sig_remover,
+        # Sign container images
+        container_signature_handler = ContainerSignatureHandler(
+            self.hub, self.task_id, self.target_settings, self.target_name
+        )
+        operator_signature_handler = OperatorSignatureHandler(
+            self.hub, self.task_id, self.target_settings, self.target_name
+        )
+        sig_remover = SignatureRemover()
+        sig_remover.set_quay_client(self.dest_quay_client)
+
+        container_signature_handler.sign_container_images(docker_push_items)
+        # Push container images
+        container_pusher = ContainerImagePusher(docker_push_items, self.target_settings)
+        container_pusher.push_container_images()
+
+        if operator_push_items:
+            # Build index images
+            operator_pusher = OperatorPusher(operator_push_items, self.target_settings)
+            existing_index_images = operator_pusher.get_existing_index_images(self.dest_quay_client)
+            iib_results = operator_pusher.build_index_images()
+            # Sign operator images
+            successful_results = dict(
+                [(key, val) for key, val in iib_results.items() if val["iib_result"]]
             )
+            operator_signature_handler.sign_operator_images(successful_results, index_stamp)
+            # Push index images to Quay
+            operator_pusher.push_index_images(successful_results, index_stamp)
```

Comment:
push_index_images has:

```
        for version in self.version_items_mapping:
            build_details = iib_results[version]["iib_result"]
```
It complains when a version is not in results.

Reply:
fixed